src/qibo/quantum_info/metrics.py

```diff
@@ -10,7 +11,7 @@
 
 
 def purity(state):
-    """Purity of a quantum state :math:`\\rho`, which is given by
+    """Purity of a quantum state :math:`\\rho`, which is given by.
```

Comment:
```suggestion
    """Purity of a quantum state :math:`\\rho`, which is given by
```
I believe it was caused by some editor extension that is automatically adding `.` at the end of the docstrings, however I don't think this works correctly in these cases since the sentence continues to the equation below.

The same applies to most docstring changes applied to this file, but I will only write here. It may be a bit tedious to fix.

Reply:
Yes, you're right, I didn't notice it.

I will immediately fix the docstrings.